js/util/verticalize_punctuation.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const scriptDetection = require('./script_detection');
+
+module.exports = function verticalizePunctuation(input) {
+    let output = '';
+
+    for (let i = 0; i < input.length; i++) {
+        const nextCharCode = input.charCodeAt(i + 1) || null;
+        const prevCharCode = input.charCodeAt(i - 1) || null;
+
+        const canReplacePunctuation = (
+            (!nextCharCode || scriptDetection.charAllowsVerticalWritingMode(nextCharCode, true)) &&
+            (!prevCharCode || scriptDetection.charAllowsVerticalWritingMode(prevCharCode, true))
+        );
+
+        if (canReplacePunctuation && module.exports.lookup[input[i]]) {
+            output += module.exports.lookup[input[i]];
+        } else {
+            output += input[i];
+        }
+    }
+
+    return output;
+};
+
+module.exports.lookup = {
+    '。': '︒',
+    '—': '︱',
+    '–': '︲',
+    '_': '︳',
+    '（': '︵',
+    '）': '︶',
+    '｛': '︷',
+    '｝': '︸',
+    '〔': '︹',
+    '〕': '︺',
+    '〘': '︹',
+    '〙': '︺',
+    '【': '︻',
+    '】': '︼',
+    '《': '︽',
+    '》': '︾',
+    '〈': '︿',
+    '〉': '﹀',
+    '「': '﹁',
+    '」': '﹂',
+    '『': '﹃',
+    '』': '﹄',
+    '｢': '﹁',
+    '｣': '﹂',
+    '［': '﹇',
+    '］': '﹈',
+    '“': '﹁',
+    '”': '﹂',
+    '‘': '﹃',
+    '’': '﹄',
+    '.': '・'
```

Comment:
@1ec5 I don't have a good sense of which punctuation should go into this table and which should not. Can you provide a table or algorithm? It may be easiest to just add a commit to the PR with your intended changes.

@nickidlugash We do not use fullwidth numerals and characters yet. Detecting when it is / isn't ok to use fullwidth characters is tricky. This can be done in a follow-up PR.